src/Mentat.Repository/Services/CardService.cs

```diff
@@ -35,9 +30,38 @@ public void RemoveCard(string id)
             _cards.DeleteOne(card => card.Id.Equals(id));
         }
 
-        public void UpdateCard(string id, Card card)
+        public void SaveCard(string id, IFormCollection collection)
```

Comment:
You have correctly identified that the controller is not the place to be fooling around with database logic and have moved this over to the CardService. Well done.

This change from the Card to an IFormCollection should be reversed. The controllers are capable of converting the IFormColleciton into a Card using a simple tag helper. Please check the controller for comments and I will tell you how.

Reply:
Definitely let the framework do the heavy lifting wherever possible (model binding and (de)serialization). It's less for us to worry about long-term.

Reply:
Committed the changes. Resolving.